nexus/types/src/inventory.rs

```diff
@@ -260,6 +260,10 @@ pub struct Caboose {
     pub git_commit: String,
     pub name: String,
     pub version: String,
+    // It is very unlikely that the sign hash will not be present.
+    // As far as we know, it will only be locally built BART signed images.
+    // We will be handling those edge cases by setting this field as an option.
```

Comment:
This isn't quite right.  As far as I've seen, standard SP images do not provide a value here.  This is probably the most specific thing I can say:

> The `sign` will generally be present for production RoT and RoT bootloader images.  It's currently absent from SP images and could be absent from RoT images as well.

but somebody else might know better.